src/mds/MDLog.cc

```diff
@@ -603,11 +614,33 @@ void MDLog::try_to_commit_open_file_table(uint64_t last_seq)
   }
 }
 
-void MDLog::trim(int m)
+void MDLog::log_trim_upkeep(void) {
+  dout(10) << dendl;
+
+  submit_mutex.lock();
+  while (!upkeep_log_trim_shutdown.load()) {
+    {
+      // adjust lock order
+      submit_mutex.unlock();
+      std::scoped_lock mlock(mds->mds_lock);
+      submit_mutex.lock();
+
+      if (mds->is_active() || mds->is_stopping()) {
+	trim();
+      }
+    }
+
+    auto now = ceph::coarse_mono_clock::now();
+    auto when = now +
+      ceph::make_timespan(g_conf().get_val<std::chrono::seconds>("mds_log_trim_upkeep_interval").count());
+    cond.wait_until(submit_mutex, when);
```

Comment:
this doesn't test for spurious wakeups - one should have a while loop testing whether we've actually reached `when` before going for the next iteration that'd call `trim`

Reply:
Generally, yes. Here, I avoided that on purpose since it isn't too bad to try trimming earlier than the configured interval (there are checks in trim() that required segments/events counts to be over a certain threshold).

Reply:
fair ask -- will update.